pycm/pycm_util.py

```diff
@@ -280,17 +281,22 @@ def transpose_func(classes, table):
     return transposed_table
 
 
-def matrix_params_from_table(table, transpose=False):
+def matrix_params_from_table(table, classes=None, transpose=False):
     """
     Calculate TP, TN, FP, and FN from the input confusion matrix.
 
     :param table: input confusion matrix
     :type table: dict
+    :param classes: ordered labels of classes
+    :type classes: list
     :param transpose: transpose flag
     :type transpose: bool
     :return: [classes_list, table, TP, TN, FP, FN]
     """
-    classes = sorted(table.keys())
+    if classes is None:
+        classes = sorted(table.keys())
+    if len(classes) < 2:
+        raise pycmMatrixError(CLASS_NUMBER_ERROR)
```

Comment:
Although we check this condition after filling the `matrix_params` later, we need the `classes` list to have more than 2 indices for the rest of this function.

Reply:
From CodeCov report I found out the outer check is no longer needed (or maybe our tests are not enough - I think in some scenarios we may need the outer check -)

Reply:
I don't think we need the outer check anymore because all scenarios will pass through `matrix_params_from_table` function.
But consider it that in direct matrix mode always ordered class list should be a subset of matrix classes (otherwise we should raise an error)